app/controllers/course/admin/codaveri_settings_controller.rb

```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 class Course::Admin::CodaveriSettingsController < Course::Admin::Controller
-  after_action :create_codaveri_problem, only: [:update_evaluator, :update_live_feedback_enabled]
+  after_action :create_or_update_codaveri_problem, only: [:update_evaluator, :update_live_feedback_enabled]
```

Comment:
Not correct behaviour, code still mass creating Codaveri problems on toggle enable (and disable) instead of creating on demand.

Reply:
the creating on-demand is indeed only done for duplication (when we duplicate question or assessment, we will set the codaveri ID to nil and then the codaveri problem will be created when user tried to attempt / get help the question). The rationale behind this is because last time we should not end up bringing the same codaveri ID to the new object

For the toggle within Admin Settings, I'm not quite convinced doing on-demand is the right thing to do; when user toggled the problem in, they are willingly converting the problem from non-codaveri to codaveri, and hence creating the codaveri problem right away is logical.

Reply:
as per discussed, we will go with the direction of creating question on-demand if the toggle for Codaveri / Live Feedback is triggered through admin panel, except if that toggle affects only 1 question (either creating immediately or updating if it's not synced with Codaveri)